memory.c

```diff
@@ -27,13 +27,22 @@
 
 void memory_protect(void)
 {
+	// Reference STM32F205 Flash programming manual revision 5 http://www.st.com/resource/en/programming_manual/cd00233952.pdf
+	// Section 2.6 Option bytes
 	//                     set RDP level 2                   WRP for sectors 0 and 1
 	if ((((*OPTION_BYTES_1) & 0xFFFF) == 0xCCFF) && (((*OPTION_BYTES_2) & 0xFFFF) == 0xFFFC)) {
 		return; // already set up correctly - bail out
 	}
-	flash_unlock_option_bytes();
```

Comment:
OK, I'll put it back. My main reason for removing it was because of the process to modify the option bytes (section 2.6.2) having the first step of waiting for any pending flash operations to complete. The libopencm3 code does the unlock after waiting, but the manual doesn't say if anything bad/unspecified will happen if that register is changed while another operation is pending. I still kind of lean towards not having the extra unlock. But, it's hard to say if it matters (though probably not).